src/cli.rs

```diff
@@ -185,47 +198,123 @@ impl Cli {
 
             match self.log_format {
                 LogFormat::Json => {
-                    tracing::subscriber::set_global_default(
-                        registry.with(
-                            tracing_subscriber::fmt::layer()
-                                .json()
-                                .with_ansi(false)
-                                .with_filter(log_filter.clone()),
-                        ),
-                    )?;
+                    if let Some(log_dir) = &self.log_dir {
+                        tracing::subscriber::set_global_default(
+                            registry
+                                .with(
+                                    tracing_subscriber::fmt::layer()
+                                        .json()
+                                        .with_ansi(false)
+                                        .with_filter(log_filter.clone()),
+                                )
+                                .with(
+                                    tracing_subscriber::fmt::layer()
+                                        .json()
+                                        .with_writer(fs::File::create(
+                                            log_dir.join("tx-proxy.log"),
+                                        )?)
+                                        .with_filter(TraceFilter),
+                                ),
+                        )?
+                    } else {
+                        tracing::subscriber::set_global_default(
```

Comment:
can this be simplified instead of having to redefine the tracing subscriber completely? Is it possible to simply add the log writing inside a conditional to an existing object